dropshot/src/test_util.rs

```diff
@@ -613,6 +613,25 @@ pub async fn objects_post<S: Serialize + Debug, T: DeserializeOwned>(
     read_json::<T>(&mut response).await
 }
 
+/**
+ * Issues an HTTP DELETE to the specified object URL to delete an object.
+ */
+pub async fn object_delete<T: DeserializeOwned>(
+    client: &ClientTestContext,
+    object_url: &str,
+) -> T {
+    let mut response = client
+        .make_request_with_body(
+            Method::DELETE,
+            &object_url,
+            "".into(),
+            StatusCode::OK,
+        )
+        .await
+        .unwrap();
+    read_json::<T>(&mut response).await
```

Comment:
... and as a follow-up, if we are sending a `NO_CONTENT` message back, it probably doesn't make sense to try to read anything from the response - it'll likely be empty.